Add platforms (osx-arm64, Windows, linux-aarch64, linux-ppc64)

ryanvolz opened the pull request:
I'm interested in adding support for the breadth of platforms that conda-forge supports, so I can use `gstreamer-orc` from `volk` which supports all of the platforms. I'm willing to volunteer as a maintainer to keep these going!

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

regro-cf-autotick-bot committed:
migration: OSXArm

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.10.1, and conda-forge-pinning 2021.05.05.20.37.49

ryanvolz committed:
Move meson to build deps, try to fix osx-arm64 cross-compilation.

ryanvolz committed:
Add default providers to build linux-aarch64 and linux-ppc64.

ryanvolz committed:
Add self to maintainers.

ryanvolz committed:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.10.1, and conda-forge-pinning 2021.05.06.10.47.03

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ryanvolz committed:
Specify libdir in same ways as ${MESON_ARGS} so it doesn't fail.

Also add the release buildtype. Add missing ninja build dep.

ryanvolz committed:
Don't fail build on unset variable, to support ${MESON_ARGS}.

In order to use ${MESON_ARGS} on all platforms to support
cross-compilation, we have to not fail when it is not set (currently on
Linux).

ryanvolz committed:
Add Windows build script and enable Windows build.

ryanvolz committed:
Don't install static libraries (CFEP 18).

ryanvolz committed:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.10.1, and conda-forge-pinning 2021.05.06.10.47.03

ryanvolz committed:
Make library tests future version proof and add test for Windows.

ryanvolz committed:
Add upstream patch to fix linux-ppc64le build.

ryanvolz committed:
Bump the build number since package contents have changed (no static).

ryanvolz committed:
Remove static library from installation on Windows.

ryanvolz committed:
Fix Windows library test.

sebastian-luna-valero reviewed (approved):
Brilliant, thank you very much and welcome to the team ;)
